### DISCUSSION TITLE
sudo

### DISCUSSION THREAD

**Discussion by 24unix** [asmvik/yabai#919] [General]
Hi, just out of curiosity:

Why does the injection into the Dock be performed via sudo?

I guess TotalSpaces also works with injection, but doesn’t need sudo.

Kind regards,
  Micha

**Comment by asmvik** [comment]
Yabai needs sudo permission because we use the mach API, in particular the `task_for_pid` and `thread_create_running` functions.
I assume TotalSpaces found a different solution for macOS Big Sur, but I'm not aware of what they did.

**Comment by 24unix** [comment]
Both projects are so great, it would be cool for all Mac users if they merge, I know, that is not an option, but they play together so nicely.

But, when I star yabai via brew services, won’t that be possible to do that as root? I guess brew services has that option.
